frontend/src/Monitor/Leaderboard/Charts/Victory.tsx

```diff
@@ -0,0 +1,102 @@
+
+// Like `Object.keys` but typed. TODO: move to reusable commons?
+type KeyValue<T extends Record<string, unknown>, K extends keyof T = keyof T> = [K, T[K]]
+export const recordEntries = <T extends Record<string, unknown>>(o: T) => Object.entries(o) as KeyValue<T>[];
+export const recordKeys = <T extends Record<string, unknown>>(o: T) => Object.keys(o) as (keyof T)[];
+export const recordValues = <T extends Record<string, unknown>>(o: T) => Object.values(o) as T[keyof T][];
+
+
+const Name = 'Victory';
+
+const useStyles = makeStyles((theme) => ({
+    formControl: {
+        margin: theme.spacing(1),
+        minWidth: 120,
+    },
+}));
+
+enum VICTORY_PARTS {
+    "Colonies" = "Colonies",
+    "Crisis Ships Killed" = "Crisis Ships Killed",
+    "Economy" = "Economy",
+    "Federation" = "Federation",
+    "Pops" = "Pops",
+    "Relics" = "Relics",
+    "Subjects" = "Subjects",
+    "Systems" = "Systems",
+    "Technology" = "Technology",
+}
+
+export const VictoryChart: React.FC<LeaderboardChartProps> = ({data, name: n, overlay}) => {
+    const classes = useStyles();
+
+    const name = n ? n : Name;
+    const {groupState, filterState} = useLeaderboardContext();
+
+    const [mode, setMode] = React.useState<'sum' | 'avg'>('avg');
+    const onModeChange = (event: React.ChangeEvent<{value: unknown}>) => setMode(event.target.value as 'sum' | 'avg');
+
+    const [parts, setParts] = React.useState<VICTORY_PARTS[]>(recordValues(VICTORY_PARTS));
+    const onPartsChange = (event: React.ChangeEvent<{value: unknown}>) => {
+        console.log(event.target.value)
+        setParts(event.target.value as VICTORY_PARTS[])
+    };
+
+    const reducer = mode === 'sum' ? sumReducer : avgReducer;
+    const selector = (snap: any, eid: number) => {
+        const obj = selectNested(`leaderboard/empire_summaries/${eid}/victory_points`, snap);
+        const sum = parts.map((part) => obj[part]).reduce(sumReducer)
+        return sum
+    };
+    const series = getTimeseries(data, groupState, filterState, selector, reducer);
+
+    return (
+        <Chart name={name} title={Name} titleColor='#6666cd' overlay={overlay}>
+            <div className='leaderboardChartForm'>
+                <FormControl className={classes.formControl}>
+                    <Select value={mode} onChange={onModeChange}>
+                        <MenuItem value='avg'>Average GDP</MenuItem>
+                        <MenuItem value='sum'>Total GDP</MenuItem>
+                    </Select>
+                </FormControl>
+            </div>
+            <div className='leaderboardChartForm'>
+                <FormControl className={classes.formControl}>
+                    <Select multiple value={parts} onChange={onPartsChange}>
+                        {recordValues(VICTORY_PARTS).map((part) => (
+                            <MenuItem key={part} value={part}>{part}</MenuItem>
+                        ))}
+                    </Select>
```

Comment:
This would probably look better and be more intuitive as a bunch of check boxes

Reply:
Due to the small amount of space, maybe it could be a menu of checkboxes that is rendered hovering over the chart and is only rendered when a button is clicked and the mouse is over it. Could add a button `Filter Categories` which opens the checkbox menu

Reply:
Check this out

![image](https://user-images.githubusercontent.com/7459419/138040843-9f0a6013-2cfe-4ef4-bbd4-09f1d3cdbb9e.png)
![image](https://user-images.githubusercontent.com/7459419/138040867-8d46130d-9b87-46a8-af82-bb66bc92371b.png)